util.js

```diff
@@ -704,12 +704,13 @@ function callbackify(original) {
     // implications (stack, `uncaughtException`, `async_hooks`)
     original.apply(this, args)
       .then(function(ret) { process.nextTick(cb.bind(null, null, ret)) },
-            function(rej) { process.nextTick(callbackifyOnRejected.bind(null, rej, cb)) });
+        function(rej) { process.nextTick(callbackifyOnRejected.bind(null, rej, cb)) });
   }
 
   Object.setPrototypeOf(callbackified, Object.getPrototypeOf(original));
-  Object.defineProperties(callbackified,
-                          getOwnPropertyDescriptors(original));
+  const desc = getOwnPropertyDescriptors(original);
+  desc.length.value += 1;
+  Object.defineProperties(callbackified, desc);
```

Comment:
The behavior depends on the NodeJS version, unmaintained versions of Node <=11 have the old (incorrect) function length behavior,  whereas the NodeJS version of node have the correct function length.


| NodeJS Version |    With NodeJS                  |    |
|------------------|----------------|--|
|   10                | callbackify((a,b,c)=>{}).length = 3 |  X |
|   11                 | callbackify((a,b,c)=>{}).length = 3 |  X |
|   12                 | callbackify((a,b,c)=>{}).length = 4 |  V |
|   14                 | callbackify((a,b,c)=>{}).length = 4 |  V |
|   16                 | callbackify((a,b,c)=>{}).length = 4 |  V |
|   18                 | callbackify((a,b,c)=>{}).length = 4 |  V |

The issue is that the polyfill is likely to be used to facilitate code sharing between modern versions of NodeJS and modern browser version, and I'm not sure that keeping the behavior of the obsolete NodeJS version would be a good idea as it would break modern code.

Reply:
I doubt much code, modern or otherwise, is checking function lengths.

It is still *very* common to support older browsers, so it's important to not try to blindly set a function's length.

Reply:
It is not set blindly.
I am following the Readme file directive:  "This module intends to provide exactly the same API as Node.js" 


> PRs are very welcome! The main way to contribute to util is by porting features, bugfixes and tests from Node.js. Ideally, code contributions to this module are copy-pasted from Node.js and transpiled to ES5, rather than reimplemented from scratch. Matching the Node.js code as closely as possible makes maintenance simpler when new changes land in Node.js. This module intends to provide exactly the same API as Node.js, so features that are not available in the core util module will not be accepted. Feature requests should instead be directed at [nodejs/node](https://github.com/nodejs/node) and will be added to this module once they are implemented in Node.js.

If there is a difference in behaviour between Node.js's util module and this module, please open an issue!

Reply:
we could argue that there was a bug in version of NodeJS < 11.0 , callbackifying a function is actually added a extra argument at the end of the function arguments that is the callback function itself , so the behavior in NodeJS>=12 is correct.

As you said, there might be not many people using the function. length functionality, so correcting the behavior to match the  correct behavior will not affect much people.

To give you an example of use:
I have code that is written for NodeJS that access either a callback function or a async function that act accordingly in the code. In order to detect how to handle it it need to know if this is a callback or async function. It does this by looking at the number of argument in the function definition. Sometime , this function is callbackified.

i.e : https://github.com/node-opcua/node-opcua/blob/263fa0ced798ef0f29436002213c03aea93ded5f/packages/node-opcua-address-space/src/ua_method_impl.ts#L127

Reply:
The correct thing to do is detect whether a function's length is configurable, and only *then* set it.

Reply:
I did this in the node test by checking thenode vesrion .
How would you suggest to do it for all browsers ?

Reply:
Thank you Jordan, I have integrated your suggestion. Can you review again ?